src/Polly.Core/Utils/Pipeline/PipelineComponentFactory.cs

```diff
@@ -14,14 +14,14 @@ internal static class PipelineComponentFactory
 
     public static PipelineComponent FromStrategy<T>(ResilienceStrategy<T> strategy) => new BridgeComponent<T>(strategy);
 
-    public static PipelineComponent WithDisposableCallbacks(PipelineComponent component, List<Action> callbacks)
+    public static PipelineComponent WithDisposableCallbacks(PipelineComponent component, IEnumerable<Action> callbacks)
     {
-        if (callbacks.Count == 0)
+        if (!callbacks.Any())
```

Comment:
Just curious on the rationale for this change (as it's internal), as the new .NET 8 analyzers for "prefer concrete types" and "don't use Any() when you have a count/length" would apply here if it actually was a list at runtime and be more performant than treating it as an enumerable.

Reply:
I wanted to do the safe copy of the list without going over that `list.ToList()` thingy :) 

Here it looks more natural.